tests/common/snappi/snappi_fixtures.py

```diff
@@ -517,3 +521,373 @@ def cvg_api(snappi_api_serv_ip,
+    l1_config = config.layer1.layer1()[-1]
+    l1_config.name = 'L1 config'
+    l1_config.port_names = [port.name for port in config.ports]
+    l1_config.speed = 'speed_{}_gbps'.format(speed_gbps)
+    l1_config.ieee_media_defaults = False
+    l1_config.auto_negotiate = False
+    l1_config.auto_negotiation.link_training = False
+    l1_config.auto_negotiation.rs_fec = True
+
+    pfc = l1_config.flow_control.ieee_802_1qbb
+    pfc.pfc_delay = 0
+    [setattr(pfc, 'pfc_class_{}'.format(i), i) for i in range(8)]
+
+    port_config_list = []
+
+    for index, duthost in enumerate(duthost_list):
+        config_result = __intf_config_multidut(
+                                                config=config,
+                                                port_config_list=port_config_list,
+                                                duthost=duthost,
+                                                snappi_ports=new_snappi_ports)
+        pytest_assert(config_result is True, 'Fail to configure Vlan interfaces')
+
+    return config, port_config_list, new_snappi_ports
+
+
+@pytest.fixture(scope="function")
+def get_multidut_snappi_ports(duthosts, conn_graph_facts, fanout_graph_facts):            # noqa: F811
+    """
+    Populate tgen ports and connected DUT ports info of T0 testbed and returns as a list
+    Args:
+        duthost (pytest fixture): duthost fixture
+        conn_graph_facts (pytest fixture): connection graph
+        fanout_graph_facts (pytest fixture): fanout graph
+    Return:
+        return tuple of duts and tgen ports
+    """
+    def _get_multidut_snappi_ports(line_card_choice, line_card_info):
+        host_names = line_card_info['hostname']
+        asic_info = line_card_info['asic']
+        asic_port_map = {
+            "asic0": ['Ethernet%d' % i for i in range(0, 72, 4)],
+            "asic1": ['Ethernet%d' % i for i in range(72, 144, 4)],
+            None: ['Ethernet%d' % i for i in range(0, 144, 4)],
+        }
+        ports = []
+        for index, host in enumerate(duthosts):
+            snappi_fanout_list = SnappiFanoutManager(fanout_graph_facts)
+            for i in range(0, 3):
+                try:
+                    snappi_fanout_list.get_fanout_device_details(i)
+                except Exception:
+                    pass
+            snappi_ports = snappi_fanout_list.get_ports(peer_device=host.hostname)
+            for port in snappi_ports:
+                port['location'] = get_snappi_port_location(port)
+                for hostname in host_names:
+                    for asic in asic_info:
+                        if port["peer_port"] in asic_port_map[asic] and hostname in port['peer_device']:
+                            port['asic_value'] = asic
+                            port['asic_type'] = host.facts["asic_type"]
+                            ports.append(port)
+        return ports
+    return _get_multidut_snappi_ports
+
+
+def get_tgen_peer_ports(snappi_ports, hostname):
```

Comment:
tgen to snappi?

Reply:
this will addressed in a separate PR , an issue has been raised to track this - > https://github.com/sonic-net/sonic-mgmt/issues/8210